source/frontend/src/features/properties/map/acquisition/detail/models.ts

```diff
@@ -47,4 +52,65 @@ export class DetailAcquisitionFileOwner {
   ownerName?: string;
   ownerOtherName?: string;
   ownerDisplayAddress?: string;
+
+  static fromApi(owner: Api_AcquisitionFileOwner): DetailAcquisitionFileOwner {
+    return {
+      ownerName: getOwnerDisplayName(owner),
+      ownerOtherName: owner.lastNameOrCorp2?.trim(),
+      ownerDisplayAddress: getFormattedAddress(owner.address),
+    };
+  }
 }
+
+const getOwnerDisplayName = (owner: Api_AcquisitionFileOwner): string => {
+  let nameDisplay = concatValues([owner.givenName, owner.lastNameOrCorp1]);
+  if (owner.incorporationNumber && owner.incorporationNumber.trim() !== '') {
+    nameDisplay = nameDisplay.concat(` (${owner.incorporationNumber})`);
+  }
+  return nameDisplay;
+};
+
+const getFormattedAddress = (address?: Api_Address): string => {
+  if (address === null || address === undefined) {
+    return '';
+  }
+
+  let addressDisplay = '';
+  let streetAddress1 = address?.streetAddress1 ? address?.streetAddress1.trim() : null;
+  let streetAddress2 = address?.streetAddress2 ? address?.streetAddress2.trim() : null;
+  let streetAddress3 = address?.streetAddress3 ? address?.streetAddress3.trim() : null;
+  let streetAddress4 = concatValues(
+    [address?.municipality, address?.province?.description, address?.postal],
+    ', ',
+  );
+
+  if (streetAddress1) {
```

Comment:
Hmm, here you are using a truthy check instead of a non-empty check. Why the difference in this context?